students/hellosputnik/main.go

```diff
@@ -0,0 +1,81 @@
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+)
+
+type Problem struct {
+	question string
+	answer   string
+}
+
+type Quiz struct {
+	problems []Problem
+	score    int
+}
+
+type Settings struct {
+	filename  *string
+	timeLimit *int
+}
+
+func main() {
+	quiz := Quiz{}
+	settings := Settings{}
+
+	settings.filename = flag.String("csv", "problems.csv", "a csv file in the format of 'question,answer'")
+	settings.timeLimit = flag.Int("limit", 30, "the time limit for the quiz in seconds")
+
+	// Get the flags (if any).
+	flag.Parse()
+
+	// Create a file handle for the file.
+	file, err := os.Open(*settings.filename)
+
+	// If there was an error opening the file, exit.
+	if err != nil {
+		panic(err)
+	}
+
+	defer file.Close()
+
+	// Create a buffered reader to read the file input.
+	fin := bufio.NewScanner(file)
+
+	// Read the problems from the comma-separated values file.
+	for fin.Scan() {
```

Comment:
You can avoid this using the package "encoding/csv" in the standard library

Reply:
👍 

@hellosputnik Generally speaking when dealing with a CSV trying to do it on your own is dangerous. Eg what happens if someone gives you the input:

```
"Yo, what is 2+2?", 4
```

From the looks of your code, this would probably end up causing issues. 

I'm guessing you didn't know about the `encoding/csv` package so it's a completely understandable (and common) mistake.